### PR TITLE
🎉 New Source: FullStory [Low code CDK] 

### PR DESCRIPTION
**Pull request by paideia1200bce**
## What
Developing new connector for the source: FullStory

Resolves:
https://github.com/airbytehq/DeveloperWeek-Connector-Contest/issues/13

## How
Developed using (Configuration Based Source) low-code CDK and connector-builder UI

## Recommended reading order
1. `spec.yaml`
2. `manifest.yaml`
3. `schemas/*`

## Tests
<strong>Integration & Acceptance</strong>
image

## 🚨 User Impact 🚨
No breaking changes, Just addition of new source

## Pre-merge Checklist
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector: FullStory</strong></summary>

### Community member or Airbyter

- [X] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [X] Secrets in the connector's spec are annotated with `airbyte_secret`
- [X] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [X] Connector's `README.md`
    - [X] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [X] `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [X] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### CODE REVIEW

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/schemas/TODO.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# TODO: Define your stream schemas
+
+Your connector must describe the schema of each stream it can output using [JSONSchema](https://json-schema.org).
+
+You can describe the schema of your streams using one `.json` file per stream.
+
+## Static schemas
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fullstory/source_fullstory/schemas/TODO.md:7]: Remove this file.

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      api_token: "{{ config['api_key'] }}"
+
+  large_value_filter:
+    type: CustomFilter
+    filter_function: |
+      def filter_records(records, stream_state, config):
+          max_state_value = stream_state.get("createdAt", "9999-04-12T18:13:36.000Z")
+          if max_state_value == "9999-04-12T18:13:36.000Z":
+              return []
+          return records
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml:18]: Why you're using this function?

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    type: DeclarativeStream
+    $parameters:
+      name: "segments"
+      path: "/segments/v1"
+    retriever:
+      $ref: "#/definitions/retriever"
+      requester:
+        $ref: "#/definitions/requester"
+        filters:
+          - $ref: "#/definitions/large_value_filter"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml:87]: https://developer.fullstory.com/server/v1/segments/list-segments/
This stream supports next token pagination, can you implement this?

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+      cursor_granularity: "PT0.000001S"
+      lookback_window: "P31D"
+      start_datetime:
+        datetime: "{{ config['start_date'] }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%S.%f%z"
+      end_datetime:
+        datetime: "{{ today_utc() }}"
+        datetime_format: "%Y-%m-%d"
+      step: "P1M"
+    primary_key: "id"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml:101]: The API doesn't support the created param in the docs. Can you share where did you find this info?

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    type: DeclarativeStream
+    $parameters:
+      name: "operations"
+      path: "/operations/v1"
+    retriever:
+      $ref: "#/definitions/retriever"
+      requester:
+        $ref: "#/definitions/requester"
+      filters:
+        - $ref: "#/definitions/large_value_filter"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml:52]: You can create a `base_stream` to not repeat yourself
```yaml
base_stream:
    type: DeclarativeStream
    retriever:
      $ref: "#/definitions/retriever"
    requester:
        $ref: "#/definitions/requester"

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments. Thanks for the contribution @amogh-opsverse

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml (hunk 2 above)
Comment (amogh-opsverse) [reply airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml:18]: it was part of a test, and is redundant, removing it.

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml (hunk 3 above)
Comment (amogh-opsverse) [reply airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml]: yeah

**Comment by amogh-opsverse** [comment]
> Left some comments. Thanks for the contribution @amogh-opsverse

Thanks for the feedback, @marcosmarxm. Btw, i'm running into this weird behavior with running the acceptance tests where they're starting to fail with the following error. The tests ran fine yesterday, could you please give me some info about this/

Screen Shot 2023-04-25 at 1 23 32 PM

**Comment by marcosmarxm** [comment]
> > Left some comments. Thanks for the contribution @amogh-opsverse
> 
> Thanks for the feedback, @marcosmarxm. Btw, i'm running into this weird behavior with running the acceptance tests where they're starting to fail with the following error. The tests ran fine yesterday, could you please give me some info about this?
> 
> Screen Shot 2023-04-25 at 1 23 32 PM

Can you push latest version of your changes?

**Comment by marcosmarxm** [comment]
Waiting contributor update the code. The version right now is not possible to reproduce issue

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml (hunk 3 above)
Comment (amogh-opsverse) [reply airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml]: implemented

> Waiting contributor update the code. The version right now is not possible to reproduce issue

@marcosmarxm, it was actually due to a formatting error while implementing the base_stream, it's resolved now, thanks. have also updated the connector with the requested changes.

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml (hunk 4 above)
Comment (amogh-opsverse) [reply airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml:82]: @marcosmarxm, i used the created field here: https://developer.fullstory.com/server/v1/segments/list-segments/

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml (hunk 5 above)
Comment (amogh-opsverse) [reply airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml]: implemented

**Comment by amogh-opsverse** [comment]
> 

this has been resolved, it was due to a formatting error; latest test results:

Screen Shot 2023-04-25 at 3 26 46 PM

**Comment by amogh-opsverse** [comment]
> Left some comments. Thanks for the contribution @amogh-opsverse

also, please let me know if i'm missing any files. I'd really appreciate it if you could help me merge this PR before the 27th deadline for the bounty.

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+  retriever:
+    type: SimpleRetriever
+    record_selector:
+      type: RecordSelector
+      extractor:
+        type: DpathExtractor
+        field_path: ["{{ parameters.name }}"]
+      paginator:
+        type: "DefaultPaginator"
+
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml:19]: Remove this extra line.

File: airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+      datetime_format: "%Y-%m-%dT%H:%M:%S.%f%z"
+      cursor_granularity: "PT0.000001S"
+      lookback_window: "P31D"
+      start_datetime:
+        datetime: "{{ config['start_date'] }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%S.%f%z"
+      end_datetime:
+        datetime: "{{ today_utc() }}"
+        datetime_format: "%Y-%m-%d"
+      step: "P1M"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fullstory/source_fullstory/manifest.yaml:53]: Remove incremental the API doesn't support it.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Remove the incremental @amogh-opsverse and ask my review again.

**Comment by marcosmarxm** [comment]
@amogh-opsverse please confirm you have granted access to maintainers to your fork.

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**